src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -1028,7 +1028,11 @@ private static void VerifyModelForOutVarInNotExecutableCode(SemanticModel model,
 
             var typeInfo = model.GetTypeInfo(decl);
             Assert.Equal(expectedType, typeInfo.Type);
-            Assert.Equal(expectedType, typeInfo.ConvertedType);
+            if (decl.IsOutDeclaration())
+            {
+                Assert.Equal(expectedType, typeInfo.ConvertedType);
```

Comment:
It looks like some of the existing tests that take advantage of this helper do not satisfy the ```decl.IsOutDeclaration()``` condition and, therefore, would lose test coverage due to this change. Could you please adjust the condition so that it only affects the scenarios where declaration expression can be converted to a different type?